MBBSEmu.Tests/ExportedModules/Majorbbs/file_Tests.cs

```diff
@@ -300,5 +300,329 @@ public void fclose_invalidPointer()
+            Assert.NotEqual(0, filep.Offset);
+
+            var stringGetPtr = mbbsEmuMemoryCore.AllocateVariable("STRING_GET", (ushort)"TEST".Length);
+            mbbsEmuMemoryCore.SetArray(stringGetPtr, Encoding.ASCII.GetBytes("TEST"));
+            
+            Assert.Equal(0, fseek(filep, 0, 2));
+
+            Assert.Equal(new FarPtr(), fgets(stringGetPtr, 4, filep));
+
+            var curFileStruct = new FileStruct(mbbsEmuMemoryCore.GetArray(filep, FileStruct.Size));
+            var curFileStream = majorbbs.FilePointerDictionary[curFileStruct.curp.Offset];
+            Assert.Equal(LOREM_IPSUM_LENGTH, curFileStream.Position );
+            Assert.True(curFileStruct.flags.IsFlagSet((ushort)FileStruct.EnumFileFlags.EOF));
+
+            Assert.Equal(0, fclose(filep));
+        }
+
+        [Fact]
+        public void fputs_fseek_EOF_file()
+        {
+            //Reset State
+            Reset();
+
+            var filePath = CreateTextFile("file.txt", LOREM_IPSUM);
+
+            Assert.Equal(LOREM_IPSUM.Length, new FileInfo(filePath).Length);
+
+            var filep = fopen("FILE.TXT", "a");
+            Assert.NotEqual(0, filep.Segment);
+            Assert.NotEqual(0, filep.Offset);
+
+            var stringPutPtr = mbbsEmuMemoryCore.AllocateVariable("STRING_PUT", (ushort)"TEST".Length);
+            mbbsEmuMemoryCore.SetArray(stringPutPtr, Encoding.ASCII.GetBytes("TEST"));
+
+            Assert.Equal(0, fseek(filep, 0, 2));
+
+            Assert.Equal(1, fputs(stringPutPtr, filep));
+
+            var curFileStruct = new FileStruct(mbbsEmuMemoryCore.GetArray(filep, FileStruct.Size));
+            var curFileStream = majorbbs.FilePointerDictionary[curFileStruct.curp.Offset];
+            Assert.Equal(LOREM_IPSUM_LENGTH + 5, curFileStream.Position);
+            Assert.True(curFileStruct.flags.IsFlagSet((ushort)FileStruct.EnumFileFlags.EOF));
+
+            Assert.Equal(0, fclose(filep));
+        }
+
+        [Fact]
+        public void fgetc_InvalidStream_Throw()
+        {
+            //Reset State
+            Reset();
+            
+            //Pass empty pointer
+            Assert.Throws<FileNotFoundException>(() => fgetc(new FarPtr()));
+        }
+
+        [Fact]
+        public void fputc_InvalidStream_Throw()
+        {
+            //Reset State
+            Reset();
+
+            //Pass empty pointer
+            Assert.Throws<FileNotFoundException>(() => fputc(0, new FarPtr()));
+        }
+
+        [Fact]
+        public void fgets_InvalidStream_Throw()
+        {
+            //Reset State
+            Reset();
+
+            //Pass empty pointer
+            Assert.Throws<FileNotFoundException>(() => fgets(new FarPtr(), 0, new FarPtr()));
+        }
+
+        [Fact]
+        public void fputs_InvalidStream_Throw()
+        {
+            //Reset State
+            Reset();
+
+            //Pass empty pointer
+            Assert.Throws<FileNotFoundException>(() => fputs(new FarPtr(), new FarPtr()));
```

Comment:
technically this should return EOF and not throw :) But that'll require mods to majorbbs - would be a good follow on

Reply:
Agree, will want to find a module that i can test with that acknowledges the filestream missing error -- will handle as a follow up